src/puppeteer.js

```diff
@@ -244,13 +238,8 @@ export const launchPuppeteer = async (options = {}) => {
     if (optsCopy.useChrome && (optsCopy.executablePath === undefined || optsCopy.executablePath === null)) {
         optsCopy.executablePath = process.env[ENV_VARS.CHROME_EXECUTABLE_PATH] || getTypicalChromeExecutablePath();
     }
-    if (optsCopy.useApifyProxy) {
-        optsCopy.proxyUrl = getApifyProxyUrl({
-            groups: optsCopy.apifyProxyGroups,
-            session: optsCopy.apifyProxySession,
-            groupsParamName: 'options.apifyProxyGroups',
-            sessionParamName: 'options.apifyProxySession',
-        });
+    if (optsCopy.proxyConfiguration) {
+        optsCopy.proxyUrl = optsCopy.proxyConfiguration.getUrl(optsCopy.apifyProxySession);
```

Comment:
This is not ideal. First of all, the `apifyProxySession` is actually `apifyProxySessionId` and second, this means that one would have to pass both `proxyConfiguration` and `apifyProxySession`, which feels complicated and hard to explain.

Let's brainstorm about this later.

Reply:
I think the naming is really the only thing here. The `apifyProxySession` is made by `sessionPool` in the `puppeteerPool` when you use `puppeteerCrawler`. There is no other usage of it anywhere if you do not use `sessionPool` in `puppeteerCrawler.` So we can just rename it to `proxySessionId` and keep it that way. 

The same n the case of `Apify.launchPuppeteer()`, we can rename it?

Reply:
Hmm, I wonder if maybe we were too eager to put `proxyConfiguration` everywhere. Maybe only the crawlers should accept it and `requestAsBrowser` and `launchPuppeteer` should only accept `proxyUrl`, because there's no way to integrate `SessionPool` with those two.